### PR TITLE
Replit Privacy Issue

### PR DESCRIPTION
**Pull request by buddy-codes**
## Description 
This PR makes sure that your private repls are not shown in your status.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
Screen shot of the presence working

image_2022-08-24_081013645

</details>

### CODE REVIEW

**Comment by buddy-codes** [comment]
it wouldn't let me `yarn lint`.

**Comment by RisingSunLight42** [comment]
First :
- Fix your PR description, the checks for the acknowledgement have to be like this`[x]`.
- Also, fix your screenshot display, an image link on Github need to have a line feed at the top,  like this :
```
Blabla bla

(link) 
```

To finish, the command is `yarn format` and not `yarn lint` ^^

File: websites/R/Replit/dist/metadata.json [hunk 1]
```diff
@@ -19,7 +23,7 @@
 		"nl": "Replit, voorheen Repl.it, is een in San Francisco gevestigde start-up en een online IDE (geïntegreerde ontwikkelomgeving) die 40 programmeertalen ondersteunt, waaronder Python, Lua, JavaScript, Julia, C++, C++11, C, C#, HTML/CSS/JS, Python met turtle en LOLCODE. Replit stelt gebruikers in staat om code te schrijven en apps te bouwen met behulp van een browser, zonder software te hoeven installeren op hun apparaten."
 	},
 	"url": "replit.com",
-	"version": "1.0.12",
+	"version": "1.0.13",
```
Comment (RisingSunLight42) [inline websites/R/Replit/dist/metadata.json:26]: It's a feature addition, so it's more a `minor` change than a `patch`
```suggestion
	"version": "1.1.0",
```

File: websites/R/Replit/presence.ts [hunk 2]
```diff
@@ -1,89 +1,91 @@
...
+        : 0;
+    presenceData.details = `Viewing repls ${length ? `(Total ${length})` : ""}`;
+    presenceData.state = `${pathname === "/repls"
+      ? "In the main page"
+      : `In a folder : ${pathname.replace("/repls/folder/", "")}`}`;
+  } else if (pathname.startsWith("/community")) {
+    presenceData.details = "Surfing feed";
+    const [postType] = pathname.replace("/community/", "").split("/"),
+      postElement: HTMLDivElement = document.querySelector(
+        "#__next > div > div.jsx-132086333.content > div.jsx-2019133593 > div.jsx-2019133593.post-page-content > div.jsx-347352367 > div.jsx-347352367.board-post-detail-header > div.jsx-347352367.board-post-detail-title"
```
Comment (RisingSunLight42) [inline websites/R/Replit/presence.ts:74]: Is that possible to shorten this query?

File: websites/R/Replit/presence.ts [hunk 3]
```diff
@@ -1,89 +1,91 @@
...
+    presenceData.details = "Browsing languages:";
+    presenceData.state = "All languages";
+  } else if (pathname.startsWith("/new")) {
+    presenceData.details = "Creating new repl:";
+    presenceData.state = `${document.querySelector<HTMLInputElement>(
+      "input[data-cy='new-repl-language-input']"
+    ).value || "No languages chosen"}`;
+  } else if (pathname.startsWith("/repls")) {
+    const repls = document.querySelector<HTMLDivElement>(
+        "#__next > div > div > div.jsx-2888589246.content > div.jsx-1264267603.repl-content > div.jsx-4064465542 > div:nth-child(3)"
```
Comment (RisingSunLight42) [inline websites/R/Replit/presence.ts:61]: Is it possible to shorten this query?

File: websites/R/Replit/presence.ts [hunk 4]
```diff
@@ -1,89 +1,91 @@
...
+        ".content span"
+      ).textContent;
+    } else {
+      let lang = document.querySelector<HTMLDivElement>(
+        "div.language-icon-container"
+      ).title;
+      let replthing = await fetch(location.href, { credentials: "omit" });
+      presenceData.details = `Editing a ${lang} repl`;
+      if (replthing.status == 404) {
+        presenceData.state = `Private Repl`;
```
Comment (RisingSunLight42) [inline websites/R/Replit/presence.ts:40]: So it's not possible to detect a private repo without fetch?

**Review by RisingSunLight42 — CHANGES_REQUESTED** [review]

**Comment by RisingSunLight42** [comment]
Also, I think it a good idea to add a `privacy` setting to let user choose if he wants to show its private repo name.

File: websites/R/Replit/presence.ts (hunk 4 above)
Comment (buddy-codes) [reply websites/R/Replit/presence.ts:40]: I tried, It seems like the div is removed from the DOM.

Comment (RisingSunLight42) [reply websites/R/Replit/presence.ts:40]: Okay, you can resolve it then

File: websites/R/Replit/presence.ts (hunk 3 above)
Comment (buddy-codes) [reply websites/R/Replit/presence.ts:61]: Don't think so.

File: websites/R/Replit/presence.ts (hunk 2 above)
Comment (buddy-codes) [reply websites/R/Replit/presence.ts:74]: Don't think so.

Comment (ImRodry) [reply websites/R/Replit/presence.ts:74]: yes it is

Comment (DarkVillager) [reply websites/R/Replit/presence.ts:74]: Doesn't even seem to work for me?

**Pull request closed without merging**